pkg/config/config_parser.go

```diff
@@ -1,47 +1,160 @@
 package config
 
 import (
+	"bytes"
+	_ "embed"
 	"errors"
 	"fmt"
+	"io"
 	"path/filepath"
 	"strings"
 
 	"github.com/cloudquery/cloudquery/internal/logging"
+	"github.com/cloudquery/cq-provider-sdk/provider/diag"
 	"github.com/hashicorp/hcl/v2"
 	"github.com/hashicorp/hcl/v2/gohcl"
 	"github.com/spf13/viper"
+	"github.com/xeipuuv/gojsonschema"
+	"gopkg.in/yaml.v3"
 )
 
-func (p *Parser) LoadConfigFromSource(name string, data []byte) (*Config, hcl.Diagnostics) {
-	if strings.HasSuffix(name, ".json") {
-		// we dropped support for json so error out with an explainable message
-		return nil, hcl.Diagnostics{{
-			Severity: hcl.DiagError,
-			Summary:  `json is not supported please use hcl format`,
-			Detail:   `json is not supported please use hcl format`,
-		}}
+//go:embed schema.json
+var configSchemaYAML []byte
+
+// configSchemaHCL is the HCL schema for the top-level of a config file. We use
+// the low-level HCL API for this level so we can easily deal with each
+// block type separately with its own decoding logic.
+var configSchemaHCL = &hcl.BodySchema{
+	Blocks: []hcl.BlockHeaderSchema{
+		{
+			Type: "cloudquery",
+		},
+		{
+			Type:       "provider",
+			LabelNames: []string{"name"},
+		},
+		{
+			Type:       "policy",
+			LabelNames: []string{"name"},
+		},
```

Comment:
don't think we allow policy blocks in config anymore.

Reply:
Yes we don't but there's a deprecation message. This was moved from another file, so when the PR to really remove it lands, this line can also be removed by that.